fireworks/core/firework.py

```diff
@@ -518,744 +269,142 @@ def reservedtime_secs(self):
             end = self.time_start if self.time_start else datetime.utcnow()
             return (end - start).total_seconds()
 
+    @property
+    @recursive_serialize
+    def launch(self):
+        launch = {}
+        launch['state'] = self.state
+        launch['state_history'] = self.state_history
+        launch['fworker'] = self.fworker
+        launch['host'] = self.host
+        launch['ip'] = self.ip
+        launch['trackers'] = self.trackers
+        launch['action'] = self.action
+        launch['launch_idx'] = self.launch_idx
+        launch['fw_id'] = self.fw_id
+        return launch
+
     @recursive_serialize
     def to_dict(self):
-        return {'fworker': self.fworker,
-                'fw_id': self.fw_id,
-                'launch_dir': self.launch_dir,
-                'host': self.host,
-                'ip': self.ip,
-                'trackers': self.trackers,
-                'action': self.action,
-                'state': self.state,
-                'state_history': self.state_history,
-                'launch_id': self.launch_id}
+        # put tasks in a special location of the spec
+        spec = self.spec
+        print([t for t in self.tasks])
+        spec['_tasks'] = [t.to_dict() for t in self.tasks]
+        m_dict = {'spec': spec, 'fw_id': self.fw_id, 'created_on': self.created_on,
+                  'updated_on': self.updated_on}
+
+        # keep export of new FWs to files clean
+        if self.state != 'WAITING':
+            m_dict['state'] = self.state
+
+        m_dict['name'] = self.name
+
+        m_dict['launch'] = self.launch
+
+        return m_dict
 
     @recursive_serialize
```

Comment:
I don't think having launch data as a sub-field of the fw dict really interferes with the user at all, and it makes sense since Firework does have a "launch" member. It also makes v1 db operations easier without making v2 operations harder. That being said, if you prefer all the members of 'launch' be root level in the fw dict, I can do that.